pkg/server/auth/auth.go

```diff
@@ -119,11 +164,16 @@ func WithAPIAuth(next http.Handler, srv *AuthServer, publicRoutes []string) http
 
 		switch method {
 		case OIDC:
-			if featureflags.Get(FeatureFlagOIDCAuth) == FeatureFlagSet {
+			if srv.oidcEnabled() {
 				// OIDC tokens may be passed by token or cookie
-				headerAuth := NewJWTAuthorizationHeaderPrincipalGetter(srv.Log, srv.verifier())
-				cookieAuth := NewJWTCookiePrincipalGetter(srv.Log, srv.verifier(), IDTokenCookieName)
-				multi.Getters = append(multi.Getters, headerAuth, cookieAuth)
+				multi.Getters = append(multi.Getters, NewJWTAuthorizationHeaderPrincipalGetter(srv.Log, srv.verifier()))
```

Comment:
Would you happen to know what this method is for?

Reply:
@ozamosi the `JWTAuthorizationHeader` one?

Reply:
Yes

Reply:
https://github.com/weaveworks/weave-gitops/blob/7be4190c671f86936ad87caa3fb788687a18f70c/pkg/server/auth/jwt.go#L48-L50

Reply:
Excellent - so that's what's used by all kinds of oidc proxy/API gateway style workflows.

What is the reason you chose not to support that for this feature?

Reply:
It does support it, the real difference in this code is that it stores the token on the principal, as well as validating the token as a JWT token.

There's definitely an argument to be made that we could update that to optionally store the token, but this is an odd feature.